leaf/extensions/tags.py

```diff
@@ -0,0 +1,351 @@
+                    description=tag_record["content"],
+                    color=discord.Color.dark_embed(),
+                ),
+                ephemeral=silent,
+            )
+            await self.bot.database.execute(
+                "UPDATE tags SET uses = $1 WHERE name = $2 and guild_id = $3;",
+                tag_record["uses"] + 1,
+                tag,
+                interaction.guild.id,
+            )
+        else:
+            await interaction.response.send_message(
+                embed=discord.Embed(
+                    description="That tag does not exist.",
+                    color=discord.Color.dark_embed(),
+                ),
+                ephemeral=silent,
+            )
+
+    @app_commands.describe(name="The tag of the newly created tag.")
+    @app_commands.command(name="create", description="Creates a new tag.")
+    async def create_tag(self, interaction: discord.Interaction, name: str) -> None:
+        tag_record = await self.bot.database.fetchrow(
+            "SELECT * FROM tags WHERE name = $1 AND guild_id = $2;",
+            name,
+            interaction.guild.id,
+        )
+
+        if not tag_record:
+            await interaction.response.send_message(
+                embed=discord.Embed(
+                    description="Please reply to this message with your tag content within 5 minutes.",
+                    color=discord.Color.dark_embed(),
+                )
+            )
+            message = await interaction.original_response()
+
+            def check(message):
+                return (
+                    message.channel == interaction.channel
+                    and message.author == interaction.user
+                )
+
+            message = await self.bot.wait_for("message", timeout=300, check=check)
+
+            await self.bot.database.execute(
+                """
+                INSERT INTO tags(name, guild_id, owner_id, content, created_at, last_edited_at, uses)
+                VALUES ($1, $2, $3, $4, DEFAULT, DEFAULT, DEFAULT);
+                """,
+                name,
+                interaction.guild.id,
+                interaction.user.id,
+                message.content,
+            )
+            await message.reply(
+                embed=discord.Embed(
+                    description="Your tag has successfully been created.",
+                    color=discord.Color.dark_embed(),
+                )
+            )
+        else:
+            await interaction.response.send_message(
+                embed=discord.Embed(
+                    description="That tag already exists.",
+                    color=discord.Color.dark_embed(),
+                )
+            )
+
+    @app_commands.describe(tag="The name of the tag to edit.")
+    @app_commands.command(name="edit", description="Edits the content of a tag.")
+    async def edit_tag(
+        self, interaction: discord.Interaction, tag: str, silent: Optional[bool] = False
```

Comment:
Remove "silent" parameter since it is not possible to reply to ephemeral messages anyway.